src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -30,38 +30,43 @@ internal static partial class JsonWriterHelper
             1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, // U+0050..U+005F
             0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, // U+0060..U+006F
             1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, // U+0070..U+007F
-        };
 
-        private static readonly StandardFormat s_hexStandardFormat = new StandardFormat('X', 4);
+            // Also include the ranges from U+0080 to U+00FF for performance to avoid UTF8 code from checking boundary.
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // U+00F0..U+00FF
+        };
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool NeedsEscapingNoBoundsCheck(byte value)
-        {
-            Debug.Assert(value <= LastAsciiCharacter);
-            return AllowList[value] == 0;
-        }
+#if BUILDING_INBOX_LIBRARY
+        private const string HexFormatString = "X4";
+#endif
 
-        private static bool NeedsEscapingNoBoundsCheck(char value)
-        {
-            Debug.Assert(value <= LastAsciiCharacter);
-            return AllowList[value] == 0;
-        }
+        private static readonly StandardFormat s_hexStandardFormat = new StandardFormat('X', 4);
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool NeedsEscaping(byte value) => value > LastAsciiCharacter || AllowList[value] == 0;
+        private static bool NeedsEscaping(byte value) => AllowList[value] == 0;
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool NeedsEscaping(char value) => value > LastAsciiCharacter || AllowList[value] == 0;
+        // Cast to (byte) for performance; avoids array bounds check.
```

Comment:
It doesn't look like this will help and is even increasing asm size (due to a couple extra moves).
[sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGuf6GkAHa4MDAJZCxAGTwYAgrjDjxLABbYo2MBhhQGAXgYAGBAHYAYgG4+3WwwAOUcQDdsupuSQMASjGwAJgDyQgA2AJ4Ayg7YQgA8wOG6AHwM8qGhEADu0uKihqlCMFkMiboA2mUwAHQAstgIAGrYoQCuMAwA1AzkALr2PBwDgzzGaCbjYxPTU7OT8zMLc9MA9CsMAKqdxjvG1dVbuxbDI4tnyxdLV+fja5vbxuR7Bw/kx9Snp+Tj39O//9cAT0FkCgXdDsZSM8IaR3p8RqCfkjgSjEaiQRiGOCHnRoTi4fCeGjiciSejydidig8VSCYTyWTGcipmjKcYAKw0jl0wks0n8hkCplY9YQpBcpA8+HCmVC5m3UUPMxcyw0E4jO7pXAQCRCMBtAIdDBqDpaIQAcxguAYADMoBB8PcdgAOYwMDA6iEWCy26COPQ26D4WJgI067DEFCbAAqFmdggghtt9sdYBNYAA1pJzaUIK0hAFNOFqurBpcbhXy1W0KXRtdq5X61da9wG22m42W43u+2bl3ewOO3N+0PRz3mx96ePp4ORzOx722RYud71QBfGyTuxbxzONweYheXMQUIMAByMBgAVwAFElNgHNmABRVBhuNowACUBTSGWyuVEcp33aXpDCMYxN0GewnFcdwOkPbxgAgE9z0va87zAB9swAfSKLInzTTQ3xadpvwMVJgI6VJZFERRlFUDQtB0PQGEIQhf0yHI8gwcoXySL9KNAgxwMgnhoL3ODPEQ5DTwvK9b3vR8LWw6BxHNSQWgIxjiI/MiKJIqiGBohQlBUdRNG0XR9DYjj/24oCDKEkSaDXIA)

![image](https://user-images.githubusercontent.com/6527137/62241393-ee393d00-b38d-11e9-8ce9-42d8a524fdb7.png)

Reply:
Thanks. I'll revert that